streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -523,16 +528,21 @@ private boolean wrappedExceptionIsIn(final Throwable throwable, final Set<Class<
     }
 
     private void handleStreamsUncaughtException(final Throwable throwable,
-                                                final StreamsUncaughtExceptionHandler streamsUncaughtExceptionHandler) {
+                                                final StreamsUncaughtExceptionHandler streamsUncaughtExceptionHandler,
+                                                final boolean skipThreadReplacement) {
```

Comment:
If we think that there will be other scenarios in the future where we want to override user's decision to skip replacing threads, then I feel this is okay. Otherwise I'm more inclined to just do it in a special condition as `decision == REPLACE THREAD && throwable instanceof MissingSourceTopicException`.

Reply:
Nevermind, discussed with @ableegoldman offline and we do want to keep it a general scheme.